### DISCUSSION TITLE
Teensy 4.1 Micropython with ulab

### DISCUSSION THREAD

**Discussion by LeandroFernandes88** [micropython/micropython#14026] [m.iMX RT / Teensy 4.x] [answered]
Hello everybody,

I'm looking for all the help I can get.
I've tried a few times and seen on some forums that it might be possible to build a Micropython build with native Ulab in the firmware. But I still haven't managed it.

If anyone has already done this, or knows where to find a link to download the HEX file with this configuration, it would help me a lot.

NOTE: I tried Circuitpython, but I had problems dealing with very small intervals and data acquisition with a high sampling rate. There were also some problems converting array to ndarray. That's why I'm trying Micropython+Ulab.

I'm sorry everyone, I'm a little new to using microcontrollers.

Thank you very much.

**Comment by jimmo** [comment, marked as answer]
@LeandroFernandes88 Yes this is definitely possible.

It's the exact same process as https://github.com/v923z/micropython-ulab?tab=readme-ov-file#stm-based-boards except you do it from `ports/mimxrt` rather than `ports/stm32` and use `BOARD=TEENSY41`.

If this process is not working the please paste commands and outputs/errors here.

(Flashing will be different too, see the instructions on the download page -- `teensy_loader_cli --mcu=imxrt1062 -v -w build-TEENSY41/firmware.hex` )

**Reply by LeandroFernandes88** [reply]
I really appreciate the help.

I was able to follow your instructions and the steps described in the path indicated.
I believe I generated the Micropython firmware .HEX file with Ulab for the Teensy4.1 board

A question that is off topic, I could get  also all support and help in: https://github.com/v923z/micropython-ulab/discussions/665

**Reply by v923z** [reply]
You can download the firmware from here: https://github.com/v923z/micropython-builder/releases
